pkg/query-service/app/preferences/model.go

```diff
@@ -0,0 +1,598 @@
+package preferences
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+
+	"github.com/jmoiron/sqlx"
+	"go.signoz.io/signoz/pkg/query-service/common"
+	"go.signoz.io/signoz/pkg/query-service/model"
+)
+
+var db *sqlx.DB
+
+type Preference struct {
+	Id           string `json:"id" db:"id"`
+	Name         string `json:"name" db:"name"`
+	DefaultValue string `json:"default_value" db:"default_value"`
+	DependsOn    string `json:"depends_on" db:"depends_on"`
+	UserScope    int    `json:"user" db:"user"`
+	OrgScope     int    `json:"org" db:"org"`
+	GroupId      string `json:"group_id" db:"group_id"`
+}
+
+type PreferenceGroup struct {
+	Id          string `json:"id" db:"id"`
+	Name        string `json:"name" db:"name"`
+	ParentGroup string `json:"parent_group" db:"parent_group"`
+}
+
+type PreferenceKV struct {
+	PreferenceId    string `json:"preference_id" db:"preference_id"`
+	PreferenceValue string `json:"preference_value" db:"preference_value"`
+}
+
+type UpdateOrgPreferenceRequest struct {
+	OrgId           string `json:"org_id"`
+	PreferenceId    string `json:"preference_id"`
+	PreferenceValue string `json:"preference_value"`
+}
+
+type AllPreferenceResponse struct {
+	GroupId     string                  `json:"group_id"`
+	GroupName   string                  `json:"group_name"`
+	Preferences []Preferences           `json:"preferences"`
+	ChildGroup  []AllPreferenceResponse `json:"child_groups"`
+}
+
+type Preferences struct {
+	Preference
+	Value string `json:"value" db:"value"`
+}
+
+func InitDB(datasourceName string) error {
+	var err error
+
+	db, err = sqlx.Open("sqlite3", datasourceName)
+
+	if err != nil {
+		return err
+	}
+
+	// create the preference group entity
+	table_schema := `CREATE TABLE IF NOT EXISTS preference_group(
+		id TEXT PRIMARY KEY NOT NULL,
+		name TEXT,
+		parent_group TEXT
+	);`
+
+	_, err = db.Exec(table_schema)
+	if err != nil {
+		return fmt.Errorf("error in creating preference group table: %s", err.Error())
+	}
+	// bootstrap the preference group entity data
+	bootstrapPreferenceGroup, fileError := fs.ReadFile(os.DirFS("../../pkg/query-service/app/preferences"), "bootstrap_preference_groups.json")
+
+	if fileError != nil {
+		return fmt.Errorf("error in reading bootstrap preference group: %s", fileError.Error())
+	}
+
+	preferenceGroups := []PreferenceGroup{}
+
+	if unmarshalErr := json.Unmarshal(bootstrapPreferenceGroup, &preferenceGroups); unmarshalErr != nil {
+		return fmt.Errorf("error in unmarshalling bootstrap preference groups: %s", unmarshalErr.Error())
+	}
+
+	for _, preferenceGroup := range preferenceGroups {
+
+		var preferenceGroupFromDB PreferenceGroup
+		query := `SELECT id FROM preference_group WHERE id=$1;`
+		err = db.Get(&preferenceGroupFromDB, query, preferenceGroup.Id)
+
+		// if the preference group is not present then insert the same
```

Comment:
How do we enable modification to existing records? For example, I want to update the old group name to something new. Should be updated in JSON or should it be done by a different means?

Reply:
it should be updated in the JSON as well for new tenants / users. i am coming up with a plan for updating it for existing users

Reply:
discussed the plan for updating as well. hence resolving this @srikanthccv